x-pack/plugins/security_solution/public/common/utils/privileges/index.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Rule } from '../../../detections/containers/detection_engine/rules';
+import * as i18n from '../../../detections/pages/detection_engine/rules/translations';
+import { isMlRule } from '../../../../common/machine_learning/helpers';
+import * as detectionI18n from '../../../detections/pages/detection_engine/translations';
+
+export const isBoolean = (obj: unknown): obj is boolean => typeof obj === 'boolean';
+
+export const canEditRuleWithActions = (
+  rule: Rule | null | undefined,
+  privileges:
+    | boolean
+    | Readonly<{
+        [x: string]: boolean;
+      }>
+): boolean => {
+  if (rule == null) {
+    return true;
+  }
+  if (rule.actions != null && rule.actions.length > 0 && isBoolean(privileges)) {
```

Comment:
nit: Could simplify with if you want:
```suggestion
  if (rule.actions?.length > 0 > 0 && isBoolean(privileges)) {
```